examples/diff.py

```diff
@@ -1,20 +1,26 @@
-#!/usr/bin/python
-# This trivial script demonstrates how to extract the unified diff for a single
-# commit in a local repository.
-#
-# Example usage:
-#  python examples/diff.py
+"""Diff.
+
+This trivial script demonstrates how to extract the unified diff for a single
+commit in a local repository.
+
+Example usage:
+    python examples/diff.py
+"""
+
+import sys
 
 from dulwich.repo import Repo
+
 from dulwich.patch import write_tree_diff
-import sys
 
-repo_path = "."
-commit_id = b"a6602654997420bcfd0bee2a0563d9416afe34b4"
 
-r = Repo(repo_path)
+REPO_PATH = "."
+COMMIT_ID = b"a6602654997420bcfd0bee2a0563d9416afe34b4"
+
+REPOSITORY = Repo(REPO_PATH)
```

Comment:
Because in the Python specification, all variables that are not placed in function or class must be capitalized. The purpose is to directly distinguish the variable and the function when other programs call or import the model, because the function of Python can be called without filling in the parameters, just like the variable.

Reply:
https://www.python.org/dev/peps/pep-0008/#global-variable-names suggests it's the same as for functions. Capitalization is for constants, which these are not - https://www.python.org/dev/peps/pep-0008/#constants

Reply:
Yes, Error messenger: `C0103: Constant name "dulwich_version_string" doesn't conform to UPPER_CASE naming style (invalid-name)`

Reply:
That's about a different variable, which is actually a constant.

Reply:
But Flake8 or Pylint If you don't think so, I think it is because it is not created in any function, but in the entire module. My suggestion is to follow the inspection sometimes, after all, in the future, everyone will consider this matter when designing.

Reply:
Yes, and they're wrong. repository is not a constant here.